daml-script/daml3/Daml/Script/Questions/Submit.daml

```diff
@@ -107,21 +113,23 @@ submitTreeMulti = submitTreeInternal
 data TrySubmit = TrySubmit with
   actAs : NonEmpty Party
   readAs : [Party]
+  disclosures : [Disclosure]
   commands : [Command]
 instance IsQuestion TrySubmit (Either SubmitError [CommandResult]) where command = "TrySubmit"
 
-trySubmitInternal : HasCallStack => [Party] -> [Party] -> Commands a -> Script (Either SubmitError a)
-trySubmitInternal actAs readAs cmds =
+trySubmitInternal : HasCallStack => [Party] -> [Party] -> [Disclosure] -> Commands a -> Script (Either SubmitError a)
+trySubmitInternal actAs readAs disclosures  cmds =
   fmap (fmap cmds.continue) $ lift $ TrySubmit with
     actAs = actAsNonEmpty actAs
     readAs = readAs
+    disclosures = disclosures
     commands = cmds.commands
 
 -- | Submit a transaction and recieve back either the result, or a `SubmitError`.
 -- In the majority of failures, this will not crash at runtime.
 trySubmit : HasCallStack => Party -> Commands a -> Script (Either SubmitError a)
-trySubmit p cmds = trySubmitInternal [p] [] cmds
+trySubmit p cmds = trySubmitInternal [p] [] [] cmds
 
 -- | Alternate version of `trySubmit` that allows specifying the actAs and readAs parties.
-trySubmitMulti : HasCallStack => [Party] -> [Party] -> Commands a -> Script (Either SubmitError a)
+trySubmitMulti : HasCallStack => [Party] -> [Party] -> [Disclosure] -> Commands a -> Script (Either SubmitError a)
 trySubmitMulti = trySubmitInternal
```

Comment:
This choice strays from the usual `Multi` pattern we have, I'd probably add more variants of the function for now, but we should look at improving the situation with the number of submit functions later.

Reply:
Same as above, I am not very happy that everybody add his own variant without a clear strategy. 
This will end up as a mess that we will have to maintain.